src/components/Frame/components/Toast/Toast.tsx

```diff
@@ -53,11 +69,26 @@ export default class Toast extends React.Component<Props, never> {
   }
 
   private triggerDismissalTimeout() {
-    const {onDismiss, duration = DEFAULT_TOAST_DURATION} = this.props;
+    const {onDismiss, duration, action} = this.props;
+
+    let timeoutDuration = duration || DEFAULT_TOAST_DURATION;
+
+    if (action && !duration) {
+      timeoutDuration = DEFAULT_TOAST_DURATION_WITH_ACTION;
+    } else if (
+      action &&
+      duration &&
+      duration < DEFAULT_TOAST_DURATION_WITH_ACTION
+    ) {
+      // eslint-disable-next-line no-console
+      console.log(
+        'Toast with action should persist for at least 10,000 milliseconds to give the merchant enough time to act on it.',
```

Comment:
This should be translated, but I'm not entirely sure how to import intl between this file and the public-facing Toast.tsx.